backend/curriculum_tracking/helpers.py

```diff
@@ -182,3 +184,20 @@ def agile_card_reviews_outstanding(user):
 def pull_request_reviews_outstanding(user):
     # TODO: only implement this once the github webhook is healthier
     return []
+
+
+def get_duration_from_course_component(course_component):
+    if course_component.agile_card.status in [
+        models.AgileCard.BLOCKED,
```

Comment:
Some things wont have cards. Expecting to have cards will mean they break some of the time. 

Just use the times. We dont need to query the database for related objects. 

Here is the code:

```
def get_duration(self):
    if self.start_time == None:
          return timedelta(0)
    if self.end_time == None:
          return timezone.now() - self.start_time
    eturn self.end_time - self.start_time
```

Notice the use of self. And the name of the function. These are intentional.